ADR/0048-movable-build-tests.md

```diff
@@ -0,0 +1,68 @@
+# 48. Attestable Build-Time Tests in Integration Service
+
+Date: 2025-04-06
+
+## Status
+
+Proposed
+
+## Context
+
+Some of the security scans run during the build pipeline take a long time to
+run. This forces users to wait longer to be notified of the results of their
+builds.  In order to receive more immediate feedback, users have requested that
+they be able to run security scans during their integration tests.
+
+## Decision
+
+We will add support for adding attestations to images from their integration
+pipelines. Test pipelines can be run and re-run independently of build
+pipelines but all pipelines will have attestations generated for them to enable
+Conforma evaluations.
+
+### Attestations
+
+Each task will be responsible for generating an attestation in Cosign. `vuln`
+attestations will be used for security scans rather than `provenance`
+attestations. Tasks will output `ARTIFACT_URI` and `ARTIFACT_DIGEST` which will
```

Comment:
Can we just use `*ARTIFACT_OUTPUTS`? https://tekton.dev/docs/chains/slsa-provenance/#artifact_outputs

Also, we can have specific types of result names if we want, i.e. `VULN_ARTIFACT_OUTPUTS` or `TEST_OUTPUT_ARTIFACTS`.

Reply:
We can, but I'm not sure what it gets us if Chains can watch TaskRuns since each TaskRun would only generate one artifact anyway. If you think its better to have `*_ARTIFACT_OUTPUTS` outputted by the PipelineRun that's a different story.

Reply:
If we use `*_ARTIFACT_OUTPUTS` then it is a clearer path if a task wants to produce multiple attestations.